controllers/instanceha/instanceha_controller.go

```diff
@@ -0,0 +1,576 @@
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			instancehav1.InstanceHAReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	configVars[*instance.Spec.OpenStackConfigMap] = env.SetValue(configMapHash)
+
+	_, secretHash, err := secret.GetSecret(ctx, helper, *instance.Spec.OpenStackConfigSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				instancehav1.InstanceHAReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				instancehav1.InstanceHASecretWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			instancehav1.InstanceHAReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			instancehav1.InstanceHAReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	configVars[*instance.Spec.OpenStackConfigSecret] = env.SetValue(secretHash)
+
+	_, secretHash, err = secret.GetSecret(ctx, helper, *instance.Spec.FencingSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				instancehav1.InstanceHAReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				instancehav1.InstanceHASecretWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			instancehav1.InstanceHAReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			instancehav1.InstanceHAReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	configVars[*instance.Spec.FencingSecret] = env.SetValue(secretHash)
+
+	// Check if *instance.Spec.InstanceHAConfigMap is present, if not create it.
+	// The first time this will produce an error b/c GetConfigMapAndHashWithName doesn't support retries.
+	_, configMapHash, err = configmap.GetConfigMapAndHashWithName(ctx, helper, *instance.Spec.InstanceHAConfigMap, instance.Namespace)
+	if err != nil {
+		cmLabels := labels.GetLabels(instance, labels.GetGroupLabel("instanceha"), map[string]string{})
+		envVars := make(map[string]env.Setter)
+		cms := []util.Template{
+			{
+				Name:               *instance.Spec.InstanceHAConfigMap,
+				Namespace:          instance.Namespace,
+				Type:               util.TemplateTypeConfig,
+				InstanceType:       instance.Kind,
+				AdditionalTemplate: map[string]string{},
+				Labels:             cmLabels,
+			},
+		}
+
+		err = configmap.EnsureConfigMaps(ctx, helper, instance, cms, &envVars)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		_, configMapHash, err = configmap.GetConfigMapAndHashWithName(ctx, helper, *instance.Spec.InstanceHAConfigMap, instance.Namespace)
+		configVars[*instance.Spec.InstanceHAConfigMap] = env.SetValue(configMapHash)
+	}
```

Comment:
I think we need to enhance this only run the current content of the condition if `k8s_errors.IsNotFound(err)`. otherwise log an error and return:
```
if err != nil {
	if k8s_errors.IsNotFound(err) {
		cmLabels := labels.GetLabels(instance, labels.GetGroupLabel("instanceha"), map[string]string{})
		envVars := make(map[string]env.Setter)
		cms := []util.Template{
			{
				Name:               *instance.Spec.InstanceHAConfigMap,
				Namespace:          instance.Namespace,
				Type:               util.TemplateTypeConfig,
				InstanceType:       instance.Kind,
				AdditionalTemplate: map[string]string{},
				Labels:             cmLabels,
			},
		}

		err = configmap.EnsureConfigMaps(ctx, helper, instance, cms, &envVars)
		if err != nil {
			return ctrl.Result{}, err
		}
		_, configMapHash, err = configmap.GetConfigMapAndHashWithName(ctx, helper, *instance.Spec.InstanceHAConfigMap, instance.Namespace)
	} else {
		// some other error ...
		r.Log.Error(err, fmt.Sprintf("could not fetch configmap %s", *instance.Spec.InstanceHAConfigMap))
		return ctrl.Result{}, err
	}
}
configVars[*instance.Spec.InstanceHAConfigMap] = env.SetValue(configMapHash)
```